addons/stock/models/stock_move_line.py

```diff
@@ -219,6 +219,9 @@ def write(self, vals):
         if self.env.context.get('bypass_reservation_update'):
             return super(StockMoveLine, self).write(vals)
 
+        if 'product_id' in vals and any(vals.get('state', ml.state) != 'draft' and vals['product_id'] != ml.product_id.id for ml in self):
+            raise UserError('Changing the product of this record is forbidden at this point.')
```

Comment:
Missing the `_()` translation function?

By the way, the wording of the error message is not clear and WILL trigger support tickets of customers asking what means *this record* and *at this point*
Can we use something like "Editing the product set on this stock move line is not allowed as some stock changes are already taken into account"?

Thanks :)

Reply:
please note that you cannot get this error issue without editing the views and removing the readonly or exporting the ml/changing the product_id/reimporting them, so you won't get more tickets before since the concerned users already fucked their inventory at this point and thus already opened tickets with "cannot unreserve more than" 

imo it's not important to have a super clear and specific error messages here, blocking the flow is important

anyway let's nim decide